### ISSUE TITLE
allow filepath.FilePath to create symlinks

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3122]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#3122|
|Type|enhancement|
|Created|2008-03-26 13:41:47Z|
|Branch|https://github.com/twisted/twisted/tree/linkto-3122|

Add a function that lets symlinks be created on platforms that support symlinks

Attachments:

* [filepath.linkTo.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e59/e59318d308cd87ad1571c2411663f9f788ae5226/9b367b5da29a9bbb6a38dd6bd4d5ade2156733a8.patch) (2344 bytes) - added by cyli on 2008-03-26 14:14:37Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3122 3122
type__enhancement enhancement
reporter__cyli cyli
priority__high high
milestone__ 
branch__branches_linkto_3122 branches/linkto-3122
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1206538907000000 1206538907000000
changetime__1206697871000000 1206697871000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli removed owner|
|-|-|

Added function `linkTo`, which works as:  `target.linkTo(linkName)`, much link `ln -s`.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [23103]) Branching to 'linkto-3122'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [23104]) Apply cyli's patch, with slight modification: add a new test for error cases,
that show a misinformation in the docstring.

Refs [#3122](https://github.com/twisted/twisted/issues/3122)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @cyli|
|-|-|

Can you had a look at the previous commit? I changed your docstring because it was slightly inaccurate. If it's OK for you, I'll probably merge this.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

Replying to [therve](#note_4):
> Can you had a look at the previous commit? I changed your docstring because it was slightly inaccurate. If it's OK for you, I'll probably merge this.

That looks good. :)  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `closed`|
|-|-|

(In [23113]) Merge linkto-3122

Author: cyli
Reviewer: therve
Fixes [#3122](https://github.com/twisted/twisted/issues/3122)

Add a method to FilePath objects to create symlinks.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
